src/_pytest/unittest.py

```diff
@@ -275,23 +276,20 @@ def addExpectedFailure(
     def addUnexpectedSuccess(
         self, testcase: "unittest.TestCase", reason: str = ""
     ) -> None:
-        self._store[unexpectedsuccess_key] = reason
+        msg = "Unexpected success"
+        if reason:
+            msg += f": {reason}"
+        strict = self.config.getini("xfail_strict")
+        self._store[xfailed_key] = Xfail(
+            reason=msg, run=True, strict=strict, raises=None
+        )
```

Comment:
adding marker will have the same effect
```python
self.add_marker(pytest.mark.xfail(reason=msg))
```

Reply:
If this works, I think it would be preferable, since then we wouldn't need to access private state of the `skipping` plugin, which some more maintainable/future proof.

Although, looking around a bit at `addSkip` and `addExpectedFailure`, they work by raising the `skip`/`xfail` exception and doing.. something.. with it. Is that relevant here?

Reply:
So `addExpectedFailure` raising  `xfail.Exception` to set excinfo and to  trigger the following condition in `pytest_runtest_makereport()`:

https://github.com/pytest-dev/pytest/blob/f6529fd3a14dc99939a372352652b443eaf38962/src/_pytest/skipping.py#L284-L287

And then `skipped` +  `wasxfail` = `xfailed`

For `addUnexpectedSuccess` I see only two ways:
1. add already evaluated `xfail` mark to `xfailed_key` so it would go [this path](https://github.com/pytest-dev/pytest/blob/master/src/_pytest/skipping.py#L296) (as it is currently implemented), or
2. mark it as `xfail` and let `skipping` do the rest (evaluate the mark etc)

Will update PR with 2nd implementation.